sympy/vector/tests/test_coordsysrect.py

```diff
@@ -299,84 +299,74 @@ def test_evalf():
-    a._connect_to_standard_cartesian('cylindrical')
-    assert a._transformation_equations() == (a.x * cos(a.y), a.x * sin(a.y), a.z)
-    assert a.lame_coefficients() == (1, a.y, 1)
-    assert a._inverse_transformation_equations() == (sqrt(a.x**2 + a.y**2),
-                            atan2(a.y, a.x), a.z)
-    a._connect_to_standard_cartesian('cartesian')
-    assert a._transformation_equations() == (a.x, a.y, a.z)
+    a = CoordSys3D('a', transformation='spherical',
+                   variable_names=["r", "theta", "phi"])
+    r, theta, phi = a.base_scalars()
+
+    assert r == a.r
+    assert theta == a.theta
+    assert phi == a.phi
+
+    raises(AttributeError, lambda: a.x)
+    raises(AttributeError, lambda: a.y)
+    raises(AttributeError, lambda: a.z)
+
+    assert a.transformation_to_parent() == (
+        r*sin(theta)*cos(phi),
+        r*sin(theta)*sin(phi),
+        r*cos(theta)
+    )
+    assert a.lame_coefficients() == (1, r, r*sin(theta))
+    assert a.transformation_from_parent_function()(x, y, z) == (
+        sqrt(x ** 2 + y ** 2 + z ** 2),
+        acos((z) / sqrt(x**2 + y**2 + z**2)),
+        atan2(y, x)
+    )
+    a = CoordSys3D('a', transformation='cylindrical',
+                   variable_names=["r", "theta", "z"])
+    r, theta, z = a.base_scalars()
+    assert a.transformation_to_parent() == (
+        r*cos(theta),
+        r*sin(theta),
+        z
+    )
+    assert a.lame_coefficients() == (1, a.theta, 1)
+    assert a.transformation_from_parent_function()(x, y, z) == (sqrt(x**2 + y**2),
+                            atan2(y, x), z)
+    a = CoordSys3D('a', transformation='cartesian')
+    assert a.transformation_to_parent() == (a.x, a.y, a.z)
     assert a.lame_coefficients() == (1, 1, 1)
-    assert a._inverse_transformation_equations() == (a.x, a.y, a.z)
+    assert a.transformation_from_parent_function()(x, y, z) == (x, y, z)
     # Variables and expressions
-    a._connect_to_standard_cartesian(((x, y, z), (x, y, z)))
-    assert a._transformation_equations() == (a.x, a.y, a.z)
+    x, y, z = symbols('x y z')
+    a = CoordSys3D('a', transformation=((x, y, z), (x, y, z)))
+    a._calculate_inv_trans_equations()
+    assert a.transformation_to_parent() == (a.x, a.y, a.z)
     assert a.lame_coefficients() == (1, 1, 1)
-    assert a._inverse_transformation_equations() == (a.x, a.y, a.z)
-    a._connect_to_standard_cartesian(((x, y, z), ((x * cos(y), x * sin(y), z))), inverse=False)
-    assert a._transformation_equations() == (a.x * cos(a.y), a.x * sin(a.y), a.z)
+    assert a.transformation_from_parent_function()(x, y, z) == (x, y, z)
+    a = CoordSys3D('a', transformation=(((x*cos(y), x*sin(y), z)), (x, y, z)))
+    assert a.transformation_to_parent() == (a.x * cos(a.y), a.x * sin(a.y), a.z)
     assert simplify(a.lame_coefficients()) == (1, sqrt(a.x**2), 1)
-    a._connect_to_standard_cartesian(((x, y, z), (x * sin(y) * cos(z), x * sin(y) * sin(z), x * cos(y))), inverse=False)
```

Comment:
We do not support now these type of `transformation`.
Here we mainly tested `_connect_to_standard_cartesian` which now is removed